### PR TITLE
Add possibility to set address for test_server 

### LINKED ISSUE #1643: add a possibility to start `test_server` on a particular port
Currently there is no way to set port for a [`test_server`](https://github.com/actix/actix-web/blob/master/test-server/src/lib.rs) to start on.
But sometimes http-server should know his port. 

Also there is a method [`local_addr`](https://docs.rs/actix-http/1.0.1/actix_http/struct.HttpServiceBuilder.html#method.local_addr) that set address which service is bound to. But `test_server` do not cares about that and code:
```rust
let srv = test_server(move || {
        HttpService::build()
            .local_addr(test_server_address) // this line does not make any changes
            .finish(map_config(
                {my_service_config},
                |_| AppConfig::default(),
            ))
            .tcp()
    });
```
looks wrong.

### PR DESCRIPTION
**Pull request by pronvis**
## PR Type
Feature

## PR Checklist
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview

Add possibility to set address for test_server 

Closes #1643

### CODE REVIEW

**Comment by robjtede** [comment]
We're not backporting anything to 2.0. For it to be considered, can you base this on the master branch, please?

**Comment by pronvis** [comment]
> We're not backporting anything to 2.0. For it to be considered, can you base this on the master branch, please?

done

**Comment by robjtede** [comment]
Just a changelog addition and if you could add a doc comment to the fn we can get this merged soon.

Edit: and a `cargo fmt`.

**Comment by pronvis** [comment]
`cargo fmt` done, but about documentation I am not sure, cause can't test it :(
```
error[E0308]: mismatched types
   --> test-server/src/lib.rs:247:9
    |
247 |         connect.await.map(|(_, framed)| framed)
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `actix_codec::framed::Framed`, found a different struct `actix_codec::framed::Framed`
    |
    = note: expected enum `std::result::Result<actix_codec::framed::Framed<_, actix_http::ws::codec::Codec>, _>`
               found enum `std::result::Result<actix_codec::framed::Framed<awc::connect::BoxedSocket, actix_http::ws::codec::Codec>, _>`
    = note: perhaps two different versions of crate `actix_codec` are being used?
```

File: test-server/src/lib.rs [hunk 1]
```diff
@@ -44,12 +44,20 @@ pub use actix_testing::*;
 /// }
 /// ```
 pub async fn test_server<F: ServiceFactory<TcpStream>>(factory: F) -> TestServer {
+    let tcp = net::TcpListener::bind("127.0.0.1:0").unwrap();
+    test_server_with_addr(tcp, factory).await
+}
+
+/// Start [`test server`](./fn.test_server.html) on a concrete Address
```
Comment (robjtede) [inline test-server/src/lib.rs:51]:
```suggestion
/// Start [`test server`](./fn.test_server.html) with existing `TcpListener`.
```

**Comment by robjtede** [comment]
you may need to run `cargo update` in your local copy, CI seems happy with that part

**Review by robjtede — APPROVED** [review]
Just a changelog entry left now

**Comment by pronvis** [comment]
Hope I did it right

File: test-server/CHANGES.md [hunk 2]
```diff
@@ -15,6 +15,7 @@
 * Replace deprecated `net2` crate with `socket2`
 * Update `base64` dependency to 0.12
 * Update `env_logger` dependency to 0.7
+* [#1645]: https://github.com/actix/actix-web/pull/1645 add possibility to set address for `TestServer`
```
Comment (robjtede) [inline test-server/CHANGES.md]: like this but move it to the unreleased section

```markdown
* add ability to set address for `TestServer` [#1645]

[#1645]: https://github.com/actix/actix-web/pull/1645 
```

**Review by robjtede — APPROVED** [review]
Cool, thanks very much for this :D

**Pull request merged**